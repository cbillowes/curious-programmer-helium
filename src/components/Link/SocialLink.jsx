import React, { Component } from "react"
import ExternalLink from "./ExternalLink"
const _ = require("lodash")

class SocialLink extends Component {
  render() {
    const { name, icon, href } = this.props

    return (
      <span
        key={`social-link-${_.kebabCase(href)}`}
        className={icon}
        title={name}>
        <ExternalLink to={href}>{this.props.children}</ExternalLink>
      </span>
    )
  }
}

export default SocialLink