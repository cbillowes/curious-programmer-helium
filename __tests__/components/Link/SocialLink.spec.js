import React from "react"
import renderer from "react-test-renderer"
import { mount } from "enzyme"
import SocialLink from "../../../src/components/Link/SocialLink"

describe("a social link component", () => {
  it("should render", () => {
    const component = <SocialLink />
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("should render contents", () => {
    const component = mount(<SocialLink name="GitHub" icon="github" href="https://github.com/cbillowes" />)
    expect(component.html()).toContain(`<span class="github" title="GitHub"><a href="https://github.com/cbillowes"`)
  })
})