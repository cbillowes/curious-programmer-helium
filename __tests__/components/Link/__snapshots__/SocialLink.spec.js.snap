// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`a social link component should render 1`] = `
<span>
  <a
    className=""
    rel="nofollow noopener noreferrer "
    target="_blank"
  />
</span>
`;
